website/assets/js/pages/contact.page.js

```diff
@@ -59,11 +59,11 @@ parasails.registerPage('contact', {
       }
       this.formData = _.clone(this.formDataToPrefillForLoggedInUsers);
     }
-    if(window.location.search){
-      window.history.replaceState({}, document.title, '/contact' );
+    if(this.primaryBuyingSituation && !this.me.primaryBuyingSituation){// only prefill form from session if buying situation not already set in database
```

Comment:
This would throw an error if the user is not logged in. We can prevent this by removing the `this.me.primaryBuyingSituation` condition and moving this block of code before the information is prefilled from the User record. 
If a user has a primaryBuyingSituation set in their session, we can use that value first, then override it if the user is logged on and has a `primaryBuyingSituation` saved.